corgidrp/darks.py

```diff
@@ -197,28 +230,59 @@ def build_trad_dark(dataset, detector_params, detector_regions=None, full_frame=
     frames = []
     bpmaps = []
     errs = []
-    for fr in dataset.frames:
-        # ensure frame is in float so nan can be assigned, though it should
-        # already be float
-        frame = fr.data.astype(float)
-        # For the fit, all types of bad pixels should be masked:
-        b1 = fr.dq.astype(bool).astype(int)
-        err = fr.err[0]
-        frame[telem_rows] = np.nan
-        i0 = slice_section(frame, 'SCI', 'image', detector_regions)
-        if np.isnan(i0).any():
-            raise ValueError('telem_rows cannot be in image area.')
-        frame[telem_rows] = 0
-        frames.append(frame)
-        bpmaps.append(b1)
-        errs.append(err)
+    if dataset[0].data is not None:
```

Comment:
because of how many times we're reading it data in this function, I suggest we use one function to do the loading each time (just to improve code reuse and code readability). I would suggest just doing `Image(filepath)`, but open to alternatives.